src/Sentry/Internal/PollingNetworkStatusListener.cs

```diff
@@ -0,0 +1,69 @@
+using Sentry.Extensibility;
+
+namespace Sentry.Internal;
+
+internal class PollingNetworkStatusListener : INetworkStatusListener
+{
+    private long _networkIsUnavailable = 0;
+
+    private readonly SentryOptions? _options;
+    private readonly IPing? _testPing;
+    internal int _delayInMilliseconds;
+    private readonly int _maxDelayInMilliseconds;
+    private readonly Func<int, int> _backoffFunction;
+
+    public PollingNetworkStatusListener(SentryOptions options, int initialDelayInMilliseconds = 500,
+        int maxDelayInMilliseconds = 32_000, Func<int, int>? backoffFunction = null)
+    {
+        _options = options;
+        _delayInMilliseconds = initialDelayInMilliseconds;
+        _maxDelayInMilliseconds = maxDelayInMilliseconds;
+        _backoffFunction = backoffFunction ?? (x => x * 2);
+    }
+
+    /// <summary>
+    /// Overload for testing
+    /// </summary>
+    internal PollingNetworkStatusListener(IPing testPing, int initialDelayInMilliseconds = 500,
+        int maxDelayInMilliseconds = 32_000, Func<int, int>? backoffFunction = null)
+    {
+        _testPing = testPing;
+        _delayInMilliseconds = initialDelayInMilliseconds;
+        _maxDelayInMilliseconds = maxDelayInMilliseconds;
+        _backoffFunction = backoffFunction ?? (x => x * 2);
+    }
+
+    private Lazy<IPing> LazyPing => new(() =>
+    {
+        if (_testPing != null)
+        {
+            return _testPing;
+        }
+        var uri = new Uri(_options!.Dsn!);
```

Comment:
`!` makes me nervous.
But I guess if we get to this point it's safe to assume it's there?

Reply:
I've added a comment to explain why this is OK.